sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -335,6 +349,29 @@ public static Task<WorkspaceStack> CreateOrSelectStackAsync(LocalProgramArgs arg
 
         private static readonly string[] SettingsExtensions = new string[] { ".yaml", ".yml", ".json" };
 
+        private async Task PopulatePulumiVersionAsync(CancellationToken cancellationToken = default)
+        {
+            var result = await this.RunCommandAsync(new[] { "version" }, cancellationToken).ConfigureAwait(false);
+            var versionString = result.StandardOutput.Trim();
+            versionString = versionString.TrimStart('v');
+            if (!SemVersion.TryParse(versionString, out var version))
+            {
+                throw new InvalidOperationException("Failed to get Pulumi version.");
+            }
+            LocalWorkspace.ValidatePulumiVersion(LocalWorkspace._minimumVersion, version);
+            this.pulumiVersion = version;
+        }
+
+        internal static void ValidatePulumiVersion(SemVersion minVersion, SemVersion currentVersion)
+        {
+            if (minVersion.Major < currentVersion.Major) {
+                throw new InvalidOperationException($"Major version mismatch. You are using Pulumi CLI version {currentVersion} with Automation SDK v{minVersion.Major}. Please update the SDK.");
+            }
+            if (minVersion.CompareTo(currentVersion) == 1) {
```

Comment:
Curious - why use CompareTo instead of operators (`<`)? Off the top of my head, you shouldn't check the result of `CompareTo` to be equal to 1 - rather being higher than zero. But operators are much easier to read.

Reply:
Thank you!